diff --git a/pkgs/development/python-modules/albumentations/default.nix b/pkgs/development/python-modules/albumentations/default.nix
index 0cf23c5bf18e..8c7aa11d1fff 100644
--- a/pkgs/development/python-modules/albumentations/default.nix
+++ b/pkgs/development/python-modules/albumentations/default.nix
@@ -75,6 +75,7 @@ buildPythonPackage rec {
 
   disabledTests = [
     "test_pca_inverse_transform"
+    "test_gaussian_blur_matches_pil"
     # these tests hang
     "test_keypoint_remap_methods"
     "test_multiprocessing_support"
