diff --git a/packages/backend/server/src/base/logger/service.ts b/packages/backend/server/src/base/logger/service.ts
index 34887d0535..894e5a578f 100644
--- a/packages/backend/server/src/base/logger/service.ts
+++ b/packages/backend/server/src/base/logger/service.ts
@@ -9,7 +9,9 @@ import { UserFriendlyError } from '../error';
 export class AFFiNELogger extends ConsoleLogger {
   constructor(context?: string) {
     super();
-    this.setLogLevels(['log', 'fatal', 'error', 'warn']);
+    this.setLogLevels((process.env.DEBUG_LOGGING === 'true') ?
+        ['log', 'fatal', 'error', 'warn', 'debug', 'verbose'] :
+        ['log', 'fatal', 'error', 'warn']);
     this.setContext(context);
   }
 
