diff --git a/packages/backend/server/src/base/logger/service.ts b/packages/backend/server/src/base/logger/service.ts
index b0b6962cc0..894e5a578f 100644
--- a/packages/backend/server/src/base/logger/service.ts
+++ b/packages/backend/server/src/base/logger/service.ts
@@ -7,6 +7,14 @@ import { UserFriendlyError } from '../error';
 // Use it via this way: `private readonly logger = new Logger(MyService.name)`
 @Injectable()
 export class AFFiNELogger extends ConsoleLogger {
+  constructor(context?: string) {
+    super();
+    this.setLogLevels((process.env.DEBUG_LOGGING === 'true') ?
+        ['log', 'fatal', 'error', 'warn', 'debug', 'verbose'] :
+        ['log', 'fatal', 'error', 'warn']);
+    this.setContext(context);
+  }
+
   override stringifyMessage(message: unknown, logLevel: LogLevel) {
     const messageString = super.stringifyMessage(message, logLevel);
     const requestId = AFFiNELogger.getRequestId();
diff --git a/packages/common/nbstore/src/impls/cloud/socket.ts b/packages/common/nbstore/src/impls/cloud/socket.ts
index 0da5776d97..1399c7bd3c 100644
--- a/packages/common/nbstore/src/impls/cloud/socket.ts
+++ b/packages/common/nbstore/src/impls/cloud/socket.ts
@@ -204,7 +204,7 @@ class SocketManager {
   constructor(endpoint: string, isSelfHosted: boolean) {
     this.socketIOManager = new SocketIOManager(endpoint, {
       autoConnect: false,
-      transports: isSelfHosted ? ['polling', 'websocket'] : ['websocket'], // self-hosted server may not support websocket
+      transports: ['websocket'],
       secure: new URL(endpoint).protocol === 'https:',
       // we will handle reconnection by ourselves
       reconnection: false,
