diff --git a/packages/backend/native/index.js b/packages/backend/native/index.js
index 5392699df6..11c3f0e50d 100644
--- a/packages/backend/native/index.js
+++ b/packages/backend/native/index.js
@@ -1,14 +1,2 @@
 /** @type {import('.')} */
-let binding;
-try {
-  binding = require('./server-native.node');
-} catch {
-  binding =
-    process.arch === 'arm64'
-      ? require('./server-native.arm64.node')
-      : process.arch === 'arm'
-        ? require('./server-native.armv7.node')
-        : require('./server-native.x64.node');
-}
-
-module.exports = binding;
+module.exports = require('./server-native.node');
