diff --git a/build.gradle b/build.gradle
index 77d3144..09f5736 100755
--- a/build.gradle
+++ b/build.gradle
@@ -21,9 +21,6 @@
 
 // This must be before plugins!
 buildscript {
-    repositories {
-      mavenCentral()
-    }
     dependencies {
         classpath("org.wisepersist:gwt-gradle-plugin:1.1.19")
     }
@@ -36,19 +33,6 @@ plugins {
     //id 'war'                                      // 
 }
 
-// https://docs.gradle.org/current/userguide/repository_types.html
-repositories {
-    //flatDir { dirs "lib" }        // use a (flat) filesystem directory as a repository
-    mavenLocal()                    // use the your local Maven cache as a repository
-    //mavenCentral()                // use MC for popular OSS Java libraries
-    //jcenter()                     // use Bintray's JCenter for Maven OSS artifacts
-    //google()                      // use Android-specific artifacts including the Android SDK
-    maven {
-        url = 'https://plugins.gradle.org/m2/'
-        url = 'https://repo.maven.apache.org/maven2'
-    }
-}
-
 dependencies {
     implementation 'com.google.gwt:gwt-user:2.9.0'
     implementation 'com.google.gwt:gwt-dev:2.9.0'
