diff --git a/build.gradle b/build.gradle
index 77d3144..09f5736 100755
--- a/build.gradle
+++ b/build.gradle
@@ -21,9 +21,6 @@
 
 // This must be before plugins!
 buildscript {
-    repositories {
-      mavenCentral()
-    }
     dependencies {
         classpath("org.wisepersist:gwt-gradle-plugin:1.1.19")
     }
@@ -36,19 +33,6 @@ plugins {
     //id 'war'                                      // 
 }
 
-// https://docs.gradle.org/current/userguide/repository_types.html
-repositories {
-    //flatDir { dirs "lib" }        // use a (flat) filesystem directory as a repository
-    mavenLocal()                    // use the your local Maven cache as a repository
-    //mavenCentral()                // use MC for popular OSS Java libraries
-    //jcenter()                     // use Bintray's JCenter for Maven OSS artifacts
-    //google()                      // use Android-specific artifacts including the Android SDK
-    maven {
-        url = 'https://plugins.gradle.org/m2/'
-        url = 'https://repo.maven.apache.org/maven2'
-    }
-}
-
 dependencies {
     implementation 'com.google.gwt:gwt-user:2.9.0'
     implementation 'com.google.gwt:gwt-dev:2.9.0'
diff --git a/docs-template/app.qmd b/docs-template/app.qmd
index 4032aea..ee7a3b8 100644
--- a/docs-template/app.qmd
+++ b/docs-template/app.qmd
@@ -6,7 +6,7 @@ toc: false
 ---
 
 <div style="width: 100%; height: 85vh; border: 1px solid #ddd;">
-<iframe src="circuitjs.html?startCircuit=1debug.txt"
+<iframe src="circuitjs.html?startCircuit=startup.txt"
         width="100%" 
         height="100%" 
         frameborder="0"
diff --git a/src/com/lushprojects/circuitjs1/public/setuplist.txt b/src/com/lushprojects/circuitjs1/public/setuplist.txt
index 76537d1..a3c1e49 100644
--- a/src/com/lushprojects/circuitjs1/public/setuplist.txt
+++ b/src/com/lushprojects/circuitjs1/public/setuplist.txt
@@ -16,6 +16,7 @@ econ_KeenMinskyGodley02.txt Keen Minsky Godley 02
 1deb_ODE2.txt Predator Prey ODE Test 2
 1debugcopy3.txt Debug 3
 1debugcopy5.txt Debug 5
+startup.txt Startup
 -
 +Basics
 ohms.txt Ohm's Law
diff --git a/war/circuitjs.html b/war/circuitjs.html
index 7f50129..92f4ce4 100644
--- a/war/circuitjs.html
+++ b/war/circuitjs.html
@@ -1,7 +1,7 @@
 <!DOCTYPE html>
 <html>
   <head>
-  <link rel="manifest" href="/circuit/manifest.json">
+  <link rel="manifest" href="/manifest.json">
   
     <!-- Properties can be specified to influence deferred binding -->
     <meta name='gwt:property' content='locale=en_UK'>
diff --git a/war/manifest.json b/war/manifest.json
index f8a79dc..793107e 100644
--- a/war/manifest.json
+++ b/war/manifest.json
@@ -1,19 +1,19 @@
 {
   "name": "CircuitJS1",
   "short_name": "CircuitJS1",
-  "start_url": "/circuit/circuitjs.html",
+  "start_url": "/circuitjs.html",
   "display": "standalone",
   "background_color": "#000000",
   "theme_color": "#2196f3",
   "description": "Circuit Simulator",
   "icons": [
     {
-      "src": "/circuit/icon512.png",
+      "src": "/icon512.png",
       "sizes": "512x512",
       "type": "image/png"
     },
     {
-      "src": "/circuit/icon1024.png",
+      "src": "/icon1024.png",
       "sizes": "128x128",
       "type": "image/png"
     }
diff --git a/war/service-worker.js b/war/service-worker.js
index b7af0a8..fd473a2 100644
--- a/war/service-worker.js
+++ b/war/service-worker.js
@@ -1,23 +1,23 @@
 const CACHE_NAME = 'circuitjs1-app-cache-v1';
 const urlsToCache = [
-  '/circuit/about.html',
-  '/circuit/canvas2svg.js',
-  '/circuit/circuitjs.html',
-  '/circuit/crystal.html',
-  '/circuit/customfunction.html',
-  '/circuit/customlogic.html',
-  '/circuit/customtransformer.html',
-  '/circuit/diodecalc.html',
-  '/circuit/icon512.png',
-  '/circuit/icon128.png',
-  '/circuit/iframe.html',
-  '/circuit/lz-string.min.js',
-  '/circuit/manifest.json',
-  '/circuit/mexle.html',
-  '/circuit/mosfet-beta.html',
-  '/circuit/opampreal.html',
-  '/circuit/split.js',
-  '/circuit/subcircuits.html',
+  '/about.html',
+  '/canvas2svg.js',
+  '/circuitjs.html',
+  '/crystal.html',
+  '/customfunction.html',
+  '/customlogic.html',
+  '/customtransformer.html',
+  '/diodecalc.html',
+  '/icon512.png',
+  '/icon128.png',
+  '/iframe.html',
+  '/lz-string.min.js',
+  '/manifest.json',
+  '/mexle.html',
+  '/mosfet-beta.html',
+  '/opampreal.html',
+  '/split.js',
+  '/subcircuits.html',
   // put everything else here
 ];
 
